infrastructure/src/main/java/org/corfudb/infrastructure/BaseServer.java

```diff
@@ -27,24 +34,42 @@
 
     private final ExecutorService executor;
 
-    /** HandlerMethod for the base server. */
     @Getter
+    @Deprecated
     private final HandlerMethods handler = HandlerMethods.generateHandler(MethodHandles.lookup(), this);
 
+    /**
+     * RequestHandlerMethods for the Base server
+     */
+    @Getter
+    private final RequestHandlerMethods handlerMethods =
+            RequestHandlerMethods.generateHandler(MethodHandles.lookup(), this);
+
     @Override
+    @Deprecated
     public boolean isServerReadyToHandleMsg(CorfuMsg msg) {
+        throw new UnsupportedOperationException("This operation is not supported");
+    }
+
+    @Override
+    public boolean isServerReadyToHandleMsg(RequestMsg request) {
         return getState() == ServerState.READY;
     }
 
     public BaseServer(@Nonnull ServerContext context) {
-        this.serverContext = context;
-        executor = Executors.newFixedThreadPool(serverContext.getBaseServerThreadCount(),
-                new ServerThreadFactory("baseServer-", new ServerThreadFactory.ExceptionHandler()));
+        serverContext = context;
+        executor = serverContext.getExecutorService(serverContext.getBaseServerThreadCount(), "baseServer-");
```

Comment:
I think the executor can be constructed and passed from the parent, but we can do those changes in another PR.